src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -551,4 +635,52 @@ private void runWithRetries(ContextualRunnable runnable, TableBlob blob, String
       throw e;
     }
   }
+
+  private long getLatestSequenceNum(TableId tableId) throws InterruptedException, DeltaFailureException {
+    RetryPolicy<Object> retryPolicy = new RetryPolicy<>()
+      .withMaxDuration(Duration.of(context.getMaxRetrySeconds(), ChronoUnit.SECONDS))
+      // maxDuration must be greater than the delay, otherwise Failsafe complains
+      .withBackoff(Math.min(61, context.getMaxRetrySeconds()) - 1, 120, ChronoUnit.SECONDS)
+      .withMaxAttempts(Integer.MAX_VALUE)
+      .onFailedAttempt(failureContext -> {
+        Throwable t = failureContext.getLastFailure();
+        LOG.error("Failed to read maximum sequence number from {}.{}.", tableId.getDataset(), tableId.getTable(), t);
+      });
+
+    try {
+      return Failsafe.with(retryPolicy).get(() -> {
+        if (bigQuery.getTable(tableId) == null) {
+          return 0L;
+        }
+
+        String query = String.format("SELECT MAX(_sequence_num) FROM %s.%s", tableId.getDataset(), tableId.getTable());
+        QueryJobConfiguration jobConfig = QueryJobConfiguration.of(query);
+        JobId jobId = JobId.of(UUID.randomUUID().toString());
+        Job queryJob = bigQuery.create(JobInfo.newBuilder(jobConfig).setJobId(jobId).build());
+        queryJob.waitFor();
+        TableResult result = queryJob.getQueryResults();
+        Iterator<FieldValueList> resultIter = result.iterateAll().iterator();
+        if (!resultIter.hasNext()) {
+          return 0L;
+        }
+        FieldValue val = resultIter.next().get(0);
+        if (val.getValue() == null) {
+          return 0L;
+        }
```

Comment:
Can we add a comment here to explain what is get(0) for? I think that is for the getting the sequence number.